core-bundle/src/Resources/contao/library/Contao/StringUtil.php

```diff
@@ -582,8 +583,53 @@ static function (array $matches) use ($arrData)
 			);
 		};
 
-		$evaluateExpression = static function ($strExpression) use ($arrData)
+		// Check if we can use the expression language or if legacy tokens have been used
```

Comment:
Should we add a comment explaining what legacy tokens are? The discussion says "non-php-variable-compatible tokens" but I have no idea what that means TBH. 🙈

Reply:
Well everything that is not compatible with a php variable? You can even read the regex we're using. E.g. you cannot have a php variable that starts with a digit etc.

Reply:
As if I knew what `\x7f-\xff` means without looking it up. 😄

Reply:
Nobody uses those anyway. It's just what the PHP parser allows to use as variables. That's why it says "php-variable-compatible".